File: Ontology/BuiltIn.cs
namespace Ontology
{
    public class BuiltIn:
        Axiom,
        IEntity
    {
        protected readonly string _iri;
        protected readonly string _localName;

        protected BuiltIn(
            string prefixIri,
            string localName
            ) : base(null)
        {
            _iri       = prefixIri + localName;
            _localName = localName;
        }

        string IEntity.Iri => _iri;

        string IEntity.LocalName => _localName;

        public override string ToString() => _localName;
    }
}

File: Web/ClientApp/src/app/Ontology/BuiltIn.ts
import { Axiom } from "./Axiom";
import { IEntity } from "./IEntity";

export class BuiltIn
    extends Axiom
    implements IEntity
{
    Iri: string;

    protected constructor(
        prefixIri       : string,
        public LocalName: string
        )
    {
        super(null);
        this.Iri = prefixIri + this.LocalName;
    }
}
